db/migrate/20240613151304_create_declarations.rb

```diff
@@ -0,0 +1,24 @@
+class CreateDeclarations < ActiveRecord::Migration[7.1]
+  def change
+    create_enum :declaration_states, %w[submitted eligible payable paid voided ineligible awaiting_clawback clawed_back]
+    create_enum :declaration_types, %w[started retained-1 retained-2 completed]
+    create_enum :declaration_state_reasons, %w[duplicate]
+
+    create_table :declarations do |t|
+      t.uuid :ecf_id, default: "gen_random_uuid()", null: false, index: true
```

Comment:
@cwrw one for a future PR, but I noticed we default some `ecf_ids` in the migrations at the moment and not others; we should probably make them consistent. I think defaulting is probably going to be the way to go since we will be uncoupled from ECF in the future and need these IDs for a time 🤔

Reply:
@ethax-ross 
The `ecf_id`'s we display in the serializers we'll need to default as the new records on NPQ only will still need to display this.  The others maybe we can remove the column after migration deployment.

Reply:
Yeah that makes sense 👍 so probably update applications and users to default and remove from courses/lead providers after migrating